library/std/src/thread/local.rs

```diff
@@ -137,6 +137,28 @@ impl<T: 'static> fmt::Debug for LocalKey<T> {
 /// # fn main() {}
 /// ```
 ///
+/// This macro also supports wrapping initializing expression in `const` block
+/// if it can be evaluated in const context:
+///
+/// ```
+/// use std::cell::Cell;
+/// thread_local! {
+///     static INIT_FLAG: Cell<bool> = const { Cell::new(false) };
+/// }
+/// ```
+///
+/// Doing so does not change the behavior, but might enable more efficient implementation
+/// of [`LocalKey`][`std::thread::LocalKey`] on supported platforms.
```

Comment:
Fair enough. Then I would narrow my concerns to the following suggestion:

```suggestion
/// Doing so might enable more efficient implementation
/// of [`LocalKey`][`std::thread::LocalKey`] on supported platforms.
```